tgui/packages/tgui/interfaces/Sleeper.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author BenLubar (https://github.com/BenLubar)
+ * @license ISC
+ */
+
+import { useBackend } from "../backend";
+import { Box, Button, Section, Knob, ProgressBar, LabeledList, Flex, Icon, TimeDisplay, HealthStat } from "../components";
+import { Window } from "../layouts";
+import { formatTime } from "../format";
+
+const damageNum = num => num <= 0 ? "0" : num.toFixed(1);
+
+export const Sleeper = (props, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    sleeperGone,
+    hasOccupant,
+    occupantStat,
+    health,
+    oxyDamage,
+    toxDamage,
+    burnDamage,
+    bruteDamage,
+    recharging,
+    rejuvinators,
+    isTiming,
+    time,
+    timeStarted,
+    timeNow,
+  } = data;
+
+  const curTime = Math.max(timeStarted
+    ? (time || 0) + timeStarted - timeNow
+    : (time || 0), 0);
+  const canInject = hasOccupant && !isTiming && !recharging && occupantStat < 2;
+
+  return (
+    <Window
+      theme="ntos"
+      width={440}
+      height={440}>
+      <Window.Content>
+        <Section title="Occupant Statistics"
+          buttons={
+            <Button
+              icon="eject"
+              align="center"
+              color="good"
+              disabled={!hasOccupant || !!isTiming}
+              onClick={() => act("eject")}>
+              {"Eject"}
+            </Button>
+          }>
+          {!hasOccupant && (sleeperGone ? "Check connection to sleeper pod." : "The sleeper is unoccupied.")}
+          {!!hasOccupant && (
+            <LabeledList>
+              <LabeledList.Item label="Status">
+                <Icon color={occupantStat > 1 ? "bad" : occupantStat === 1 ? "average" : "good"}
+                  name={occupantStat === 0 ? "check" : occupantStat === 1 ? "bed" : "skull"} />
+                {
+                  occupantStat === 0 ? " Conscious"
+                    : occupantStat === 1 ? " Unconscious"
+                      : occupantStat === 2 ? " Dead"
+                        : (" Error " + occupantStat)
+                }
```

Comment:
Lot of `occupantStat` magic number usage here. Not expecting it to be made as a util or anything, but having a map for these things at the start of this file would probably help readability here. Something akin to:
```
const OccupantStatus = {
    Conscious: 0,
    Unconscious: 1,
  Dead: 2,
};

const occupantStatuses = {
  [OccupantStatus.Conscious]: {
    name: 'Conscious',
    color: 'good',
    icon: 'check',
  },
  [OccupantStatus.Unconscious]: {
    name: 'Unconscious',
    color: 'average',
    icon: 'bed',
  },
  [OccupantStatus.Dead]: {
    name: 'Dead',
    color: 'bad',
    icon: 'skull',
  },
};
```

Then here you could derive their status before the `return`:
```
const occupantStatus = occupantStatuses[occupantStat];
```
And rewrite this chunk as:
```suggestion
                <Icon
                  color={occupantStatus.color}
                  name={occupantStatus.icon}
                />
                {occupantStatus.name}
```
If `>` checks are needed (like earlier in this function) you can still use `occupantStat` as you have it.